numpy/lib/function_base.py

```diff
@@ -1320,6 +1320,477 @@ def gradient(f, *varargs, **kwargs):
     else:
         return outvals
 
+def divergence(v,*varargs,**kwargs):
+    """
+    Return the divergence of an N-dimensional vector field of N 
+    components, each of dimension N.
+
+    The divergence is computed using second order accurate central
+    differences in the interior and either first differences or second
+    order accurate one-sides (forward or backwards) differences at the
+    boundaries. The returned gradient hence has the same shape as the
+    input array.
+    
+    Parameters
+    ----------  
+    v : list of numpy arrays
```

Comment:
Hi,
Thanks for your suggestion. By reviewing the size test, I have realized that they are incomplete right now, they only check that every component has the same number of dimensions, but not the same size. A quick example:
>>> a = np.array([[1,2],[3,4]])
>>> b = np.array([[5,6,7],[8,9,10],[11,12,13]])
>>> c=[a,b]
>>> d = np.asanyarray(c) 
>>> np.shape(d[0])
(2,2)
>>> np.shape(d[1])
(3,3)
This should fail the tests for size. But the tests right now would pass since:
>>> len(d[0].shape)
2
>>> len(d[1].shape)
2

I'm going to fix this and will implement your suggestion as well. 
Thank you